backend/src/openbeheer/utils/decorators.py

```diff
@@ -44,6 +46,14 @@ def wrapper(*args: Params.args, **kwargs: Params.kwargs) -> Response:
                 status=504,
             )
             return Response(data, status=504)
+        except ImproperlyConfigured as e:
+            data = ExternalServiceError(
+                title="Configuration error",
+                detail=str(e),
+                code="configuration_error",
+                status=503,  # Service Unavailable
```

Comment:
Mhmmm 🤔 I am wondering if configuration error is the right error, shouldn't we just say "Service unavailable"?

If there is a configuration error, the health checks should flag it

Reply:
I agree that the health checks should give more insight:
> Perhaps we should point towards the health checks too? 🤔

I figured the status already says "Service unavailable" (because "our" service is unavailable). And I wanted to use code, title and detail to give the end user just enough insight into "why is the service unavailable", without overwhelming them; in order to differentiate this cause from one like OZ (or whatever backend service) being down.

These error shouldn't occur at all, but when it occurs IMHO the message should have information to help narrow it down. "You're holding it wrong." vs "The network is having issues."

Reply:
Ah mhmm, in my head this was the decorator for functions talking to external services. So if our BFF is unavailable, then this shouldn't be reached? Shouldn't the frontend handle if our BFF is down?

Reply:
You're correct, this decorator is the top level around `get`, `post`, `patch` etc. We should probably rename it, because this addition makes it handle exceptions of a different nature:

The BFF is not "down", but it's not in a state that it can provide its service :) I'm open to suggestions for a different status code.